File: vscode-extension/test/context.test.ts
/**
 * 上下文服务测试
 */

import { describe, it, expect } from 'vitest';

// 模拟上下文类型
interface Context {
    files: Array<{ path: string; language: string; content: string }>;
    cursor: { line: number; column: number };
    selectedCode: string | null;
}

describe('ContextService', () => {
    describe('detectLanguage', () => {
        it('should detect java', () => {
            const languageId = 'java';
            expect(['java']).toContain(languageId);
        });
        
        it('should detect python', () => {
            const languageId = 'python';
            expect(['python']).toContain(languageId);
        });
        
        it('should detect javascript', () => {
            const languageId = 'javascript';
            expect(['javascript']).toContain(languageId);
        });
        
        it('should detect typescript', () => {
            const languageId = 'typescript';
            expect(['typescript']).toContain(languageId);
        });
    });
    
    describe('Context structure', () => {
        it('should create context with files', () => {
            const context: Context = {
                files: [{ path: 'test.java', language: 'java', content: 'public class Test {}' }],
                cursor: { line: 1, column: 1 },
                selectedCode: null,
            };
            
            expect(context.files).toHaveLength(1);
            expect(context.files[0].language).toBe('java');
        });
        
        it('should create context with cursor', () => {
            const context: Context = {
                files: [],
                cursor: { line: 42, column: 15 },
                selectedCode: null,
            };
            
            expect(context.cursor.line).toBe(42);
            expect(context.cursor.column).toBe(15);
        });
        
        it('should create context with selected code', () => {
            const context: Context = {
                files: [],
                cursor: { line: 1, column: 1 },
                selectedCode: 'const x = 1;',
            };
            
            expect(context.selectedCode).toBe('const x = 1;');
        });
    });
});

File: idea-plugin/bin/test/com/vex/coding/ContextTest.kt
package com.vex.coding.service

import org.junit.jupiter.api.Test
import org.assertj.core.api.Assertions.assertThat

/**
 * ContextService 单元测试
 */
class ContextTest {
    
    @Test
    fun `should create context with files and cursor`() {
        val context = Context(
            files = listOf(
                FileContext(
                    path = "test.java",
                    language = "java",
                    content = "public class Test {}"
                )
            ),
            cursor = CursorPosition(1, 1),
            selectedCode = null
        )
        
        assertThat(context.files).hasSize(1)
        assertThat(context.cursor.line).isEqualTo(1)
        assertThat(context.cursor.column).isEqualTo(1)
    }
    
    @Test
    fun `should create context with selected code`() {
        val context = Context(
            files = emptyList(),
            cursor = CursorPosition(10, 5),
            selectedCode = "const x = 1;"
        )
        
        assertThat(context.selectedCode).isEqualTo("const x = 1;")
    }
    
    @Test
    fun `should create file context`() {
        val fileContext = FileContext(
            path = "src/main/java/Test.java",
            language = "java",
            content = "public class Test {}"
        )
        
        assertThat(fileContext.path).contains("Test.java")
        assertThat(fileContext.language).isEqualTo("java")
        assertThat(fileContext.content).contains("class Test")
    }
    
    @Test
    fun `should create cursor position`() {
        val cursor = CursorPosition(42, 15)
        
        assertThat(cursor.line).isEqualTo(42)
        assertThat(cursor.column).isEqualTo(15)
    }
}